include/envoy/config/incremental_subscription.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "envoy/common/exception.h"
+#include "envoy/common/pure.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Config {
+
+template <class ResourceType> class IncrementalSubscriptionCallbacks {
+public:
+  typedef Protobuf::RepeatedPtrField<ResourceType> ResourceVector;
+
+  virtual ~IncrementalSubscriptionCallbacks() {}
+
+  /**
+   * Called when an incremental configuration update is received.
+   * @param added_resources resources newly added since the previous fetch.
+   * @param removed_resources names of resources that this fetch instructed to be removed.
+   * @param version_info update version.
+   * @throw EnvoyException with reason if the incremental config changes are rejected. Otherwise the
+   *        changes are accepted. Accepted changes have their version_info reflected in subsequent
+   *        requests.
+   */
+  virtual void
+  onIncrementalConfig(const Protobuf::RepeatedPtrField<envoy::api::v2::Resource>& added_resources,
```

Comment:
Nit: prefer consistency in naming with non-incremental; either `onIncrementalConfigUpdate` or `onConfigUpdate` (since this is already under an incremental namespace).

Reply:
Done.